src/common/mfu_daos.c

```diff
@@ -1313,3 +1330,2195 @@ int mfu_flist_copy_daos(
+    }
+    hdf5->usr_attr = H5Acreate2(hdf5->file,
+                                prop_str,
+                                hdf5->attr_dtype,
+                                hdf5->attr_dspace,
+                                H5P_DEFAULT,
+                                H5P_DEFAULT);
+    if (hdf5->usr_attr < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create attribute");
+        rc = 1;
+        goto out;
+    }   
+    status = H5Awrite(hdf5->usr_attr, hdf5->attr_dtype,
+                      attr_val);
+    if (status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to write attribute");
+        rc = 1;
+        goto out;
+    }
+    status = H5Aclose(hdf5->usr_attr);
+    if (status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to close attribute");
+        rc = 1;
+        goto out;
+    }
+
+out:
+    return rc;
+}
+
+static int cont_serialize_prop_uint(struct hdf5_args *hdf5,
+                                    uint64_t *attr_val,
+                                    char *prop_str)
+{
+    int     rc = 0;
+    hid_t   status = 0;
+
+    hdf5->attr_dims[0] = 1;
+    hdf5->attr_dtype = H5Tcopy(H5T_NATIVE_UINT64);
+    status = H5Tset_size(hdf5->attr_dtype, 8);
+    if (status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create version dtype");
+        rc = 1;
+        goto out;
+    }
+    if (hdf5->attr_dtype < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create usr attr type");
+        rc = 1;
+        goto out;
+    }
+    hdf5->attr_dspace = H5Screate_simple(1, hdf5->attr_dims,
+                                         NULL);
+    if (hdf5->attr_dspace < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create version attr dspace");
+        rc = 1;
+        goto out;
+    }
+    hdf5->usr_attr = H5Acreate2(hdf5->file,
+                                prop_str,
+                                hdf5->attr_dtype,
+                                hdf5->attr_dspace,
+                                H5P_DEFAULT,
+                                H5P_DEFAULT);
+    if (hdf5->usr_attr < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create attr");
+        rc = 1;
+        goto out;
+    }   
+    status = H5Awrite(hdf5->usr_attr, hdf5->attr_dtype,
+                      attr_val);
+    if (status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to write attr");
+        rc = 1;
+        goto out;
+    }
+    status = H5Aclose(hdf5->usr_attr);
+    if (status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to close attr");
+        rc = 1;
+        goto out;
+    }
+
+out:
+    return rc;
+}
+
+static int cont_serialize_props(struct hdf5_args *hdf5,
+                                daos_handle_t cont)
+{
+    int                     rc = 0;
+    daos_prop_t             *prop_query;
+    uint32_t                i;
+    char                    cont_str[64];
+    int                     len = 0;
+
+    /*
+     * Get all props except the ACL first.
+     */
+    prop_query = daos_prop_alloc(1);
+    if (prop_query == NULL)
+        return ENOMEM;
```

Comment:
Another mfu style thing is to always use braces, even for one-liners.
```suggestion
    if (prop_query == NULL) {
        return ENOMEM;
    }
```